repository/githubapi/git.go

```diff
@@ -0,0 +1,41 @@
+package githubapi
+
+import (
+	"context"
+
+	"release-manager/repository/githubapi/model"
+	"release-manager/repository/githubapi/util"
+	svcmodel "release-manager/service/model"
+
+	"github.com/google/go-github/v60/github"
+)
+
+const (
+	tagsToFetch = 100
+)
+
+// GitRepository is used to access all git objects in the git repository hosted on GitHub
+type GitRepository struct{}
+
+func NewGitRepository() *GitRepository {
+	return &GitRepository{}
+}
+
+func (c *GitRepository) ReadTags(ctx context.Context, tkn string, repo svcmodel.GithubRepo) ([]svcmodel.GitTag, error) {
+	client := util.GetGithubClient(tkn)
+
+	// Up to 100 tags can be fetched per page
+	// If the number of pages is not specified in the list options, only one page will be fetched
+	// https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repository-tags
+	t, _, err := client.Repositories.ListTags(
+		ctx,
+		repo.OwnerSlug,
+		repo.RepositorySlug,
+		&github.ListOptions{PerPage: tagsToFetch},
+	)
+	if err != nil {
+		return nil, util.ToResponseError(err)
```

Comment:
And as you mentioned @bolshakoff, here simply makes sense to return `responseerror`.

Reply:
I see. Yeah, I have "to an api layer", which is not precise, because I meant service layer.
Sorry for a confusion.

So, in general, the principle is:
- domain / svc knows only about itself and its errors
- repo translates to domain errors
- transport translates from domain errors to transport errors

Reply:
What you had before, i.e. translation from `GithubError` to `APIERror` in svc wasn't too bad.
But as you can see, it becomes too many imports, too many external knowledge for the domain. But domain should ideally know only about itself, so that you always have a guarantee that HERE you are dealing purely with the business logic.
So, for now, I'd move it back.

Reply:
>But as you can see, it becomes too many imports, too many external knowledge for the domain.

So, as you can see, the approach where it's up to other layers to translate to/from domain/svc layer might be pleasant.
As a result, there is a simple "sun" topology, where everybody imports domain models and domain errors.. everybody serves to domain.. and that's fine, because for all repositories, 3rd party services and transport layer it's just one import, as opposed to the situation when domain/service layer might have to import tons of different modules and deal with tons of different types of errors...